docs/design/004-payment-interface/README.md

```diff
@@ -740,4 +741,121 @@ pub extern fn indy_parse_verify_payment_response(command_handle: i32,
                                                  cb: Option<extern fn(command_handle_: i32,
                                                                       err: ErrorCode,
                                                                       txn_json: *const c_char)>) -> ErrorCode {}
-```
\ No newline at end of file
+
+/// Signs a message with a payment address.
+///
+/// #Params
+/// command_handle: command handle to map callback to user context.
+/// wallet_handle: wallet handler (created by open_wallet).
+/// address: payment address of message signer. The key must be created by calling indy_create_address
+/// message_raw: a pointer to first byte of message to be signed
+/// message_len: a message length
+/// cb: Callback that takes command result as parameter.
+///
+/// #Returns
+/// a signature string
+///
+/// #Errors
+/// Common*
+/// Wallet*
+/// Crypto*
+#[no_mangle]
+pub extern fn indy_sign_with_address(command_handle: CommandHandle,
+                                     wallet_handle: WalletHandle,
+                                     address: *const c_char,
+                                     message_raw: *const u8,
+                                     message_len: u32,
+                                     cb: Option<extern fn(command_handle_: CommandHandle,
+                                                          err: ErrorCode,
+                                                          signature_raw: *const u8,
+                                                          signature_len: u32)>) -> ErrorCode {
+    trace!("indy_sign_with_address: >>> wallet_handle: {:?}, address: {:?}, message_raw: {:?}, message_len: {:?}",
```

Comment:
in order to match the rest of the documentation, the method signature is all that should follow the description of the inputs/outputs